tree/ntuple/v7/src/RField.cxx

```diff
@@ -769,52 +775,58 @@ ROOT::Experimental::RClassField::RClassField(std::string_view fieldName, std::st
    if (fClass->Property() & kIsDefinedInStd) {
       throw RException(R__FAIL(std::string(className) + " is not supported"));
    }
-   TIter next(fClass->GetListOfDataMembers());
-   while (auto dataMember = static_cast<TDataMember *>(next())) {
-      //printf("Now looking at %s %s\n", dataMember->GetName(), dataMember->GetFullTypeName());
+
+   for (auto baseClass : ROOT::Detail::TRangeStaticCast<TBaseClass>(*fClass->GetListOfBases())) {
+      TClass *c = baseClass->GetClassPointer();
+      auto subField = Detail::RFieldBase::Create(std::string(kPrefixInherited) + c->GetName(),
```

Comment:
I think for the moment it's fine to use the type name as field name.  Together with the v1 serialization, I'd like to go through the RField classes and change the use of type names altogether to generic names (e.g. `R_0`, `R_1`, etc.). This way, field names are valid C++ identifiers (except for the `:` of base classes).